File: seqkit/duplicate.go
package seqkit

import "ignis/driver/api"

type SeqKitDuplicateOptions struct {
	inner DuplicateOptions
}

type DuplicateOptions struct {
	Config KitConfig
	Times  *int64
}

func (this *DuplicateOptions) setDefaults() *DuplicateOptions {
	this.Config.setDefaults()
	setDefault(&this.Times, 1)

	return this
}

func (this *SeqKitDuplicateOptions) Config(v *SeqKitConfig) *SeqKitDuplicateOptions {
	this.inner.Config = v.inner
	return this
}

func (this *SeqKitDuplicateOptions) Times(v int64) *SeqKitDuplicateOptions {
	this.inner.Times = &v
	return this
}

func Duplicate(input *api.IDataFrame[string], o *SeqKitDuplicateOptions) (*api.IDataFrame[string], error) {
	if o == nil {
		o = &SeqKitDuplicateOptions{}
	}
	opts := o.inner
	opts.setDefaults()

	dup, err := api.AddParam[int64](libSource("Duplicate"), "times", *opts.Times)
	if err != nil {
		return nil, err
	}

	return api.Flatmap[string, string](input, dup)
}

File: bigseqkit-py/bigseqkit/duplicate.py
from bigseqkit.helper import _setDefault, _libSource, _config, _optionsToString, _parseKargs, SeqKitConfig, IDataFrame


class SeqKitDuplicateOptions:

    def __init__(self):
        self.__inner = DuplicateOptions()

    def config(self, v: SeqKitConfig):
        self.__inner.Config = _config(v)

    def times(self, v: int):
        self.__inner.Times = v

    def _run(self, input: IDataFrame, **kwargs):
        opts = self.__inner
        _parseKargs(opts, kwargs)
        opts.setDefaults()
        libprepare = _libSource("Duplicate").addParam("times", opts.Times)
        return input.mapPartitions(libprepare)

class DuplicateOptions:

    def __init__(self):
        self.Config = None  # KitConfig
        self.Times = None  # int

    def setDefaults(self):
        _setDefault(self, "Config", _config(SeqKitConfig())).setDefaults()
        _setDefault(self, "Times", 1)


def duplicate(input: IDataFrame, o: SeqKitDuplicateOptions = None, **kwargs):
    if o is None:
        o = SeqKitDuplicateOptions()
    return o._run(input, **kwargs)
